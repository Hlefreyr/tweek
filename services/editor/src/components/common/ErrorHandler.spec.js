/* global jest, before, beforeEach, describe, it, expect */
import React from 'react';
import { render } from '@testing-library/react';
import ErrorHandler from './ErrorHandler';

const ErrorComponent = () => {
  throw 'error';
};

describe('ErrorHandler component', () => {
  it('should not render when exception is thrown and no error message is passed', () => {
    const { asFragment } = render(
      <ErrorHandler>
        <ErrorComponent />
      </ErrorHandler>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error message when exception is thrown', () => {
    const { asFragment } = render(
      <ErrorHandler errorMessage="some error message">
        <ErrorComponent />
      </ErrorHandler>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onError when exception is thrown', () => {
    const onErrorMock = jest.fn();
    render(
      <ErrorHandler onError={onErrorMock}>
        <ErrorComponent />
      </ErrorHandler>,
    );

    expect(onErrorMock).toHaveBeenCalled();
  });

  it('should render original component correctly', () => {
    const onErrorMock = jest.fn();

    const { asFragment } = render(
      <ErrorHandler errorMessage="some error message" onError={onErrorMock}>
        <div>some element</div>
        <div>another element</div>
      </ErrorHandler>,
    );

    expect(onErrorMock).not.toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
