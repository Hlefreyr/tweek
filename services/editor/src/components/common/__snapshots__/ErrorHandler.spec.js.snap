// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler component should not render when exception is thrown and no error message is passed 1`] = `<DocumentFragment />`;

exports[`ErrorHandler component should render error message when exception is thrown 1`] = `
<DocumentFragment>
  <div>
    some error message
  </div>
</DocumentFragment>
`;

exports[`ErrorHandler component should render original component correctly 1`] = `
<DocumentFragment>
  <div>
    some element
  </div>
  <div>
    another element
  </div>
</DocumentFragment>
`;
